Project1/src/World.cpp

```diff
@@ -175,28 +176,47 @@ void World::Collisions() {
 		// Check that the tile it's moving into is not a solid tile
 		if (std::max(overlapLeft, overlapRight) <= std::max(overlapTop, overlapBottom))
 		{
-			// Collision on right side
-			if (overlapLeft <= overlapRight && !(it-Height)->Solid)
-			{
-				player->Pos.x = tile.Pos.x - player->Bounds.x;
+			// Collision on right or left side
+
+			auto newIterator = it-Height;
+			std::shared_ptr<Tile> rightTile;
+
+			if (tile->Pos.x == 0) {
+				player->Pos.x = tile->Pos.x + tile->Bounds.x;

```

Comment:
This is a weird hack and I'm not sure why it's necessary, this is the same as line 200. Should be able to change the if/else branch there to do the right thing.